generator/src/lib.rs

```diff
@@ -1583,19 +1583,14 @@ pub fn derive_setters(_struct: &vkxml::Struct) -> Option<Tokens> {
     }
 
     let next_function = if has_next {
-        if is_next_const {
-            quote! {
-                pub fn next<T>(mut self, next: &'a T) -> #name_builder<'a> where T: #extends_name {
-                    self.inner.p_next = next as *const T as *const c_void;
-                    self
-                }
-            }
-        } else {
-            quote! {
-                pub fn next<T>(mut self, next: &'a mut T) -> #name_builder<'a> where T: #extends_name {
-                    self.inner.p_next = next as *mut T as *mut c_void;
-                    self
+        quote! {
+            pub fn next<T: #extends_name>(mut self, next: &'a mut impl ::std::ops::DerefMut<Target = T>) -> #name_builder<'a> {
+                unsafe{
+                    let ptr = &mut self.inner as *mut _ as *mut c_void;
+                    let last_extension = ExtensionChain::last_chain(ptr);
+                    (*last_extension).p_next = next.deref_mut() as *mut T as *mut c_void;
```

Comment:
I kinda like the idea of prepending to the chain rather than appending, since it means we don't have to rely on the optimizer to erase the loop. On the other hand, the resulting sequence of links is less intuitive, and I'm not certain that the ordering never matters.